core/insertion_marker_manager.js

```diff
@@ -408,9 +408,9 @@ Blockly.InsertionMarkerManager.prototype.shouldReplace_ = function() {
   // Dragging a block over an existing block in an input.
   if (local.type == Blockly.OUTPUT_VALUE) {
     // Insert the dragged block into the stack if possible.
-    if (!closest.isConnected() ||
-      Blockly.Connection.lastConnectionInRow(this.topBlock_,
-          closest.targetConnection.getSourceBlock())) {
+    if (closest &&
+        local.getSourceBlock()
+            .shouldInsertDraggedBlock(this.topBlock_, closest)) {
```

Comment:
What if we just gave the insertion marker manager a pointer to the renderer?  Passing these all the way through the block svg seems wrong.

Reply:
Yep, that is indeed a round about way of getting there. I don't like this either. 
I just realized there was already a reference to the workspace on the insertion marker manager. Will up to go directly to the renderer.